homeassistant/components/metoffice/config_flow.py

```diff
@@ -15,30 +19,44 @@
 from homeassistant.helpers import config_validation as cv
 
 from .const import DOMAIN
-from .helpers import fetch_site
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, str]:
+async def validate_input(
+    hass: HomeAssistant, latitude: float, longitude: float, api_key: str
+) -> dict[str, Any]:
     """Validate that the user input allows us to connect to DataPoint.
 
     Data has the keys from DATA_SCHEMA with values provided by the user.
     """
-    latitude = data[CONF_LATITUDE]
-    longitude = data[CONF_LONGITUDE]
-    api_key = data[CONF_API_KEY]
-
-    connection = datapoint.connection(api_key=api_key)
+    errors = {}
+    connection = datapoint.Manager.Manager(api_key=api_key)
+
+    try:
+        forecast = await hass.async_add_executor_job(
+            connection.get_forecast,
+            latitude,
+            longitude,
+            "daily",
+            False,
+        )
 
-    site = await hass.async_add_executor_job(
-        fetch_site, connection, latitude, longitude
-    )
+        if forecast is None:
+            errors["base"] = "cannot_connect"
 
-    if site is None:
-        raise CannotConnect
+    except (HTTPError, APIException) as err:
+        if isinstance(err, HTTPError) and err.response.status_code == 401:
+            errors["base"] = "invalid_auth"
+        else:
+            errors["base"] = "cannot_connect"
+    except Exception:
+        _LOGGER.exception("Unexpected exception")
+        errors["base"] = "unknown"
+    else:
+        return {"site_name": forecast.name, "errors": errors}
 
-    return {"site_name": site.name}
+    return {"errors": errors}
```

Comment:
```suggestion
    return errors
```
Returning a dict inside the error dict seems not needed

Reply:
This is not just an error dict. This function also returns a site name which is used in to pre-populate device name